### PR TITLE
[Issue 45] Vytvorit prehlad kreditov

### PR DESCRIPTION
**Pull request by filek**
Ešte sa čaká na konzultáciu od Sama a Lukáša, po dohode s Dušanom som urobil PR.
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/Nexteria/Nextis/pull/104%23discussion_r85974615%22%5D%2C%20%22comments%22%3A%20%7B%22Pull%20d915499e11c7c08e63956f315039b34612967b00%20frontend/src/browser/users/ActivityPointsPage.js%20329%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/Nexteria/Nextis/pull/104%23discussion_r85974615%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Cele%20mi%20to%20pride%20ako%20prilis%20velka%20komponenta%2C%20rozbil%20by%20som%20to%20na%20mensie%20casti%2C%20minimalne%20na%20tie%20dve%20tabulky%22%2C%20%22created_at%22%3A%20%222016-11-01T17%3A11%3A17Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/2537379%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/ddeath%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20frontend/src/browser/users/ActivityPointsPage.js%3AL1-352%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>
- [ ] <a href='#crh-comment-Pull d915499e11c7c08e63956f315039b34612967b00 frontend/src/browser/users/ActivityPointsPage.js 329'></a> &nbsp;<b><a href='https://github.com/Nexteria/Nextis/pull/104#discussion_r85974615'>File: frontend/src/browser/users/ActivityPointsPage.js:L1-352</a></b>
- <a href='https://github.com/ddeath'></a> Cele mi to pride ako prilis velka komponenta, rozbil by som to na mensie casti, minimalne na tie dve tabulky

<a href='https://www.codereviewhub.com/Nexteria/Nextis/pull/104?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/Nexteria/Nextis/pull/104?approve=1'></a>&nbsp;<a href='https://github.com/Nexteria/Nextis/pull/104'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by ddeath** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/Nexteria/Nextis/pull/104'>Refresh</a>.<a href='#crh-update-none'></a>

File: app/NxEventAttendee.php [hunk 1]
```diff
@@ -63,4 +63,21 @@ public function attendeesGroup()
     {
         return $this->belongsTo('App\AttendeesGroup', 'attendeesGroupId');
     }
+
+    public function event()
+    {
+        $result = [];
```
Comment (ddeath) [inline app/NxEventAttendee.php:69]: toto mozes napisat takto:

```
return $this->attendeesGroup->nxEvent;
```

File: app/User.php [hunk 2]
```diff
@@ -147,4 +148,31 @@ public function hostedEvents()
...
+    public function eventAttendees()
+    {
+        return $this->hasMany('App\NxEventAttendee', 'userId');
+    }
+
+    public function computeActivityPoints()
+    {
+        $sumGainedPoints = 0;
+        $sumPotentialPoints = 0;
+        $eventAttendees = $this->eventAttendees()->get();
```
Comment (ddeath) [inline app/User.php:161]: toto mozes napisat:

```
 $eventAttendees = $this->eventAttendees;
```

ak tam nedas zatvorky, vrati to collection, ak tam das zatvorky mas stale query

File: app/User.php [hunk 3]
```diff
@@ -147,4 +148,31 @@ public function hostedEvents()
...
+        foreach ($eventAttendees as $attendee) {
+            $event = $attendee->event();
+            if($event != null)
```
Comment (ddeath) [inline app/User.php:164]: tu mozes vynechat != null

```
if ($event)
```

File: frontend/src/browser/users/ActivityPointsPage.js [hunk 4]
```diff
@@ -0,0 +1,351 @@
...
+                  </div>
+                </div>
+              </div>
+            </div>
+          </div>
+        </div>
+      </div>
+
+
+
```
Comment (ddeath) [inline frontend/src/browser/users/ActivityPointsPage.js]: prazdne medzery na zvys

File: frontend/src/browser/components/Icon.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+  static propTypes = {
+    type: PropTypes.string,
+  };
+
+  render() {
+    const { type } = this.props;
+    return (
+      <span
+        className={`fa fa-${type}`}
+        style={{ color: type == 'close' ? 'red' : 'green' }}
```
Comment (ddeath) [inline frontend/src/browser/components/Icon.js:13]: hmm toto sa mi moc nepozdava. Cize ikona moze byt iba cervena a zelena?

Mozno namiesto "type" by som tam dal ked tak "color"

File: frontend/src/browser/users/ActivityPointsPage.js [hunk 6]
```diff
@@ -0,0 +1,351 @@
...
+                        </tbody>
+                      </table>
+                    }
+                  </div>
+                </div>
+              </div>
+            </div>
+          </div>
+        </div>
+      </div>
```
Comment (ddeath) [inline frontend/src/browser/users/ActivityPointsPage.js]: Cele mi to pride ako prilis velka komponenta, rozbil by som to na mensie casti, minimalne na tie dve tabulky

**Review by ddeath — CHANGES_REQUESTED** [review]
Nejake postrehy som napisal, urcite to treba este prebehnut cez lintre, jedna php ale aj js

**Comment by Player1os** [comment]
Bug: Admin ma v prehlade https://files.slack.com/files-pri/T2AHMLX1U-F2T1Z1KRU/nexteria_space.clipular__3_.png0 bodov z NaN % (toto je aj v testovacom prostredi)

**Comment by Player1os** [comment]
V nastaveniach uzivatela je teraz na spodku minimalny pocet aktivity bodov https://files.slack.com/files-pri/T2AHMLX1U-F2T29E0DV/nexteria_space.clipular__4_.png.

Toto ma vyznam len pre Nexteria Studentov, vieme to spravit tak aby to bolo len pre nich?

V buducnosti by sme chceli aby sa defaultny minimalny pocet kreditov odvijal od toho v ktorom leveli su ale na to potom spravime samostatny issue

**Comment by Player1os** [comment]
Rozdelil by som stlpec https://files.slack.com/files-pri/T2AHMLX1U-F2T1Z1KRU/nexteria_space.clipular__3_.png v tomto obrazku na dva. Jeden nech ma celkovy pocet ziskanych kreditov "Bodov (% / možných)" na dva jeden nech je "Celkovy pocet bodov" a druhy nech je "% z minimalneho poctu".

Dovod preco to chceme je lebo chceme niekedy vediet zoradovat uzivatelov podla tychto cisel (podla % zvlast a podla poctu bodov zvlast)

**Comment by Player1os** [comment]
S dandou sme sa zatial zhodli ze ten spodny panel kde je zoznam ostatnych uzivatel moze vadit niektorym studentom (teda ze budu chiet aby ostatni videli ako sa im dari), ak na tom Mato obzvlast netrval (este nemam jeho stanovsiko) dal by som to radsej prec
vsetky eventy
kompromis by bolo tu cast kodu odkomentovat pre pripadne buduce pouzitie

**Pull request merged**